src/libraries/System.Formats.Cbor/src/System/Formats/Cbor/Writer/CborWriter.cs

```diff
@@ -205,11 +205,19 @@ private void EnsureWriteCapacity(int pendingCount)
                 throw new OverflowException();
             }
 
-            if (_buffer is null || _buffer.Length - _offset < pendingCount)
+            int currentCapacity = _buffer is null ? 0 : _buffer.Length;
+            int requiredCapacity = _offset + pendingCount;
+            if (currentCapacity < requiredCapacity)
             {
-                const int BlockSize = 1024;
-                int blocks = checked(_offset + pendingCount + (BlockSize - 1)) / BlockSize;
-                Array.Resize(ref _buffer, BlockSize * blocks);
+                int newCapacity = currentCapacity == 0 ? 1024 : currentCapacity * 2;
+                if ((uint)newCapacity > (uint)Array.MaxLength)
+                    newCapacity = Array.MaxLength;
+                if (newCapacity < requiredCapacity)
+                    newCapacity = requiredCapacity;
+
+                byte[] newBuffer = new byte[newCapacity];
```

Comment:
would it be worth using `GC.AllocateUninitializedArray` here?

Reply:
I'll let the maintainers decide whether it makes sense here.

Reply:
AllocateUninitializedArray would probably be OK, except it's not available for .NET Standard 2.0.  I don't think it's important enough here that it's worth a `#if`, so just always using `new byte[newCapacity]` seems fine.

Reply:
Why not just use Array.Resize as before?

Reply:
> Presumably the original buffer is close to full so we're not saving a lot by only copying the written slice.

The intent was to avoid copying empty space here, since copying is really the biggest cost on resizing here.